src/agent.h

```diff
@@ -382,12 +382,17 @@ class Agent : public StateWrangler, virtual public Ider {
   /// the agent has never been built).
   inline const int enter_time() const { return enter_time_; }
 
-  /// Sets the number of time steps this agent operates between building and
+  ///Sets the number of time steps this agent operates between building and
   /// decommissioning (-1 if the agent has an infinite lifetime).  This should
   /// generally only be called BEFORE an agent is added to a context as a
   /// prototype.  Throws ValueError if the agent has already been deployed.
   void lifetime(int n_timesteps);
 
+  /// Sets the number of time steps this agent operates between building and
+  /// decommissioning (-1 if the agent has an infinite lifetime).  The boolean
+  /// allows for an agent to override the ValueError.
+  void lifetime_force(int n_timesteps);
```

Comment:
Since this is likely to happen after the agent has been deployed and operated for some number of timesteps, this argument is the sum of already (completed time steps) + (remaining time steps).  What will happen if `n_timesteps` < (completed time steps)?  Will that trigger decommissioning?  Or will we have missed the chance for that?

Reply:
Yeah, it is possible that it will miss the chance to decommission. I guess I can build in a catch to ensure that it can't be set for historic time steps.